runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -483,6 +483,88 @@ bool J9::ValuePropagation::transformIndexOfKnownString(
    return false;
    }
 
+bool J9::ValuePropagation::transformUnsafeCopyMemoryCall(TR::Node *arraycopyNode)
+   {
+   if (!canRunTransformToArrayCopy())
+      return false;
+
+   TR::TreeTop *tt = _curTree;
+   TR::Node *ttNode = tt->getNode();
+
+   if (comp()->canTransformUnsafeCopyToArrayCopy()
+         && arraycopyNode->isUnsafeCopyMemoryIntrinsic())
+      {
+
+      if ((ttNode->getOpCodeValue() == TR::treetop || ttNode->getOpCode().isResolveOrNullCheck())
+            && performTransformation(comp(), "%sChanging call Unsafe.copyMemory [%p] to arraycopy\n", OPT_DETAILS, arraycopyNode))
+
+         {
+         TR::Node *unsafe     = arraycopyNode->getChild(0);
+         TR::Node *src        = arraycopyNode->getChild(1);
+         TR::Node *srcOffset  = arraycopyNode->getChild(2);
+         TR::Node *dest       = arraycopyNode->getChild(3);
+         TR::Node *destOffset = arraycopyNode->getChild(4);
+         TR::Node *len        = arraycopyNode->getChild(5);
+
+         int64_t srcOffLow;
+         int64_t srcOffHigh;
+         int64_t dstOffLow;
+         int64_t dstOffHigh;
+         int64_t copyLenLow;
+         int64_t copyLenHigh;
```

Comment:
We can move these definitions down to where the variables get set, so:

```
         int64_t srcOffLow   = srcOffsetConstraint ? srcOffsetConstraint->getLowInt() : TR::getMinSigned<TR::Int32>();
         int64_t srcOffHigh  = srcOffsetConstraint ? srcOffsetConstraint->getHighInt() : TR::getMaxSigned<TR::Int32>();
         int64_t dstOffLow   = dstOffsetConstraint ? dstOffsetConstraint->getLowInt() : TR::getMinSigned<TR::Int32>();
         int64_t dstOffHigh  = dstOffsetConstraint ? dstOffsetConstraint->getHighInt() : TR::getMaxSigned<TR::Int32>();
         int64_t copyLenLow  = copyLenConstraint   ? copyLenConstraint->getLowInt() : TR::getMinSigned<TR::Int32>();
         int64_t copyLenHigh = copyLenConstraint   ? copyLenConstraint->getHighInt() : TR::getMaxSigned<TR::Int32>();
```

Reply:
Fixed in https://github.com/eclipse/openj9/compare/cdb436ce300395af11b2c1e141c4aa783f6e71b2..d8f5f521fc955e5cb1fbfeaa0d418cb475503a1a